### PR TITLE
Use ruff for linting python files

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Use ruff for linting python files.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/__init__.py [hunk 1]
```diff
@@ -241,3 +241,5 @@
     "MlflowClient",
     "MlflowException",
 ] + _model_flavors_supported
+
+assert (1, 2, 3)
```
Comment (harupy) [inline mlflow/__init__.py:245]: Dummy change.

File: pyproject.toml [hunk 2]
```diff
@@ -9,3 +9,58 @@ force-exclude = '''
...
 '''
+
+[tool.ruff]
+line-length = 100
+target-version = "py38"
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+select = [
+  "E",
+  "F",
+]
```
Comment (harupy) [inline pyproject.toml:20]: ruff has more rules. We can incrementally add them.

File: mlflow/__init__.py (hunk 1 above)
Comment (harupy) [reply mlflow/__init__.py:245]: Problem matcher works fine.

File: mlflow/models/evaluation/validation.py [hunk 3]
```diff
@@ -59,10 +59,10 @@ def __init__(
         greater_is_better=None,
         higher_is_better=None,
     ):
-        if threshold is not None and not type(threshold) in {int, float}:
+        if threshold is not None and type(threshold) not in {int, float}:
             raise MetricThresholdClassException("`threshold` parameter must be a number.")
         if min_absolute_change is not None and (
-            not type(min_absolute_change) in {int, float} or min_absolute_change <= 0
+            type(min_absolute_change) not in {int, float} or min_absolute_change <= 0
```
Comment (harupy) [inline mlflow/models/evaluation/validation.py:65]: Made a a few change to make ruff happy.

File: .github/workflows/autoformat.yml
Comment (harupy) [inline .github/workflows/autoformat.yml:1]: /ai review the following change.

Comment (harupy) [inline .github/workflows/autoformat.yml:1]: /ai review the following change:

Comment (harupy) [inline .github/workflows/autoformat.yml:1]: /ai review this change:

**Review by gabrielfu — APPROVED** [review]

File: pylintrc [hunk 4]
```diff
@@ -66,132 +66,18 @@ confidence=
...
 # either give multiple identifier separated by comma (,) or put this option
 # multiple time (only on the command line, not in the configuration file where
 # it should appear only once). See also the "--disable" option for examples.
-enable=c-extension-no-member,
+enable=signature-differs,
        line-too-long,
+       unused-variable,
+       undefined-variable,
+       unused-import,
+       unused-argument,
```
Comment (harupy) [inline pylintrc:80]:
```suggestion
```

File: mlflow/__init__.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # pylint: disable=wrong-import-position
 """
-The ``mlflow`` module provides a high-level "fluent" API for starting and managing MLflow runs.
+THE ``mlflow`` module provides a high-level "fluent" API for starting and managing MLflow runs.
```
Comment (harupy) [inline mlflow/__init__.py:3]:
```suggestion
The ``mlflow`` module provides a high-level "fluent" API for starting and managing MLflow runs.
```

**Pull request merged**